### PR TITLE
Get GitHub file contributors during getStaticProps, cache to avoid rate limit

### PR DESCRIPTION
**Pull request by wackerow**
## Description
Updates fetching for GitHub contributors for each page, removing the client-side fetch, replacing with cached solution during getStaticProps (server-side), removing all error/loading conditionals allowing the component the render immediately.

- Deprecates `useClientSideGitHub*` hooks
- Adds new `fetchGitHistory.ts` script, designed to be run server-side during getStaticProps
- Implements caching solution to avoid repeating API calls for non-English paths, using in-memory object from the `[...slug].tsx]` page component that stores already fetched data.
- This data is now fetched in `getStaticProps`, and passed forward. Updated consuming components to remove all conditionals, as this data should never be in a loading or error state anymore.
- Patch: Removed the `isDesktop` boolean inside FileContributors and replaced with `hideBelow="md"` to fix the initial patch render glitch where this component was empty to start.

## Preview URL
- https://deploy-preview-12942--ethereumorg.netlify.app/en/developers/docs/transactions/

### CODE REVIEW

File: src/layouts/Docs.tsx [hunk 1]
```diff
@@ -206,6 +204,7 @@ type DocsLayoutProps = Pick<
   | "lastUpdatedDate"
   | "crowdinContributors"
   | "contentNotTranslated"
+  | "gitContributors"
```
Comment (wackerow) [inline src/layouts/Docs.tsx:207]: Note that `gitContributors` and `creowdinContributors` will be deprecated in #12969 in favor of a single list of `contributors`

File: src/components/FileContributors.tsx [hunk 2]
```diff
@@ -8,13 +8,11 @@ import {
   ListItem,
   ModalBody,
   ModalHeader,
-  SkeletonText,
```
Comment (pettinarip) [inline src/components/FileContributors.tsx:11]: bye bye 💀 oh much cleaner now

File: src/lib/api/fetchGitHistory.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  while (true) {
+    const response = await fetch(url.href, {
+      headers: { Authorization: `token ${gitHubToken}` },
+    })
+
+    if (
+      response.status === 403 &&
+      response.headers.get("X-RateLimit-Remaining") === "0"
+    ) {
+      const resetTime = response.headers.get("X-RateLimit-Reset") as string
```
Comment (pettinarip) [inline src/lib/api/fetchGitHistory.ts:30]: Nice use of headers for this, TIL

**Review by pettinarip — APPROVED** [review]
LGMT! good job @wackerow

**Pull request merged**